spark/src/test/java/org/apache/iceberg/spark/source/TestForwardCompatibility.java

```diff
@@ -192,8 +197,7 @@ public void testSparkCanReadUnknownTransform() throws IOException {
       manifestWriter.close();
     }
 
-    TableOperations ops = ((HasTableOperations) table).operations();
-    ops.commit(ops.current(), ops.current().replaceCurrentSnapshot(new FakeSnapshot(manifestWriter.toManifestFile())));
+    table.newFastAppend().appendManifest(manifestWriter.toManifestFile()).commit();
```

Comment:
This test needed to change because the update started catching that the manifest didn't have a snapshot ID for entries to inherit. The solution is to append the manifest like normal instead of using the commit API directly. To be able to commit the manifest without rewriting it (which fails on the fake partition data), this sets snapshot ID inheritance to true.